packages/nvidia_nat_mcp/src/nat/plugins/mcp/client_impl.py

```
@@ -152,18 +158,24 @@ async def mcp_client_function_handler(config: MCPClientConfig, builder: Builder)
-        "streamable-http": lambda: MCPStreamableHTTPClient(str(config.server.url)),
-    }.get(config.server.transport)
+    # Resolve auth provider if specified
+    auth_provider = None
+    if config.server.auth_provider:
+        auth_provider = await builder.get_auth_provider(config.server.auth_provider)
 
-    if not client_cls:
+    # Build the appropriate client
+    if config.server.transport == "stdio":
+        if not config.server.command:
+            raise ValueError("command is required for stdio transport")
```

Ruff (0.12.2)

Avoid specifying long messages outside the exception class

(TRY003)